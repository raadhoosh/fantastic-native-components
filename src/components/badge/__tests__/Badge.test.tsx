import Badge from "../index";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

const createTestProps = (props: Object) => ({
  ...props,
});

describe("Badge", () => {
  describe("rendering", () => {
    let component: ShallowWrapper;
    let props: Object;
    beforeEach(() => {
      props = createTestProps({
        color: '#f00',
        primary: true,
        secondary: true,
        success: true,
      });
      component = shallow(<Badge {...props} 
        onPress={() => {
          alert('You tapped the Badge!');
      }}
      >Badge</Badge>);
    });
    it("should render a <Badge />", () => {
      const component = shallow(<Badge {...props} >Badge</Badge>);
      expect(component).toMatchSnapshot();
    });
  });
});
