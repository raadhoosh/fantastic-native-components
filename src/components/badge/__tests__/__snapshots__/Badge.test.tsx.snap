// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge rendering should render a <Badge /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    color="#f00"
    primary={true}
    secondary={true}
    success={true}
  >
    Badge
  </Badge>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef
        color="#f00"
        primary={true}
        secondary={true}
        success={true}
      >
        Badge
      </ForwardRef>,
      "color": "#f00",
      "primary": true,
      "secondary": true,
      "success": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Badge",
        "color": "#f00",
        "primary": true,
        "secondary": true,
        "success": true,
      },
      "ref": null,
      "rendered": "Badge",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "    
    flex: 1;
    align-items: center;
    justify-content: center;   
    color: ",
            [Function],
            "; 
    font-size:",
            [Function],
            "; 
    height: 100%;    
",
          ],
        },
        "displayName": "Styled(Component)",
        "propTypes": Object {
          "accessible": [Function],
          "adjustsFontSizeToFit": [Function],
          "allowFontScaling": [Function],
          "disabled": [Function],
          "ellipsizeMode": [Function],
          "minimumFontScale": [Function],
          "nativeID": [Function],
          "numberOfLines": [Function],
          "onLayout": [Function],
          "onLongPress": [Function],
          "onPress": [Function],
          "pressRetentionOffset": [Function],
          "selectable": [Function],
          "selectionColor": [Function],
          "style": [Function],
          "suppressHighlighting": [Function],
          "testID": [Function],
          "textBreakStrategy": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "      
    background-color: ",
          [Function],
          ";    
    border: 1px solid ",
          [Function],
          ";        
    width: ",
          [Function],
          ";  
    height: ",
          [Function],
          ";  
    border-radius: ",
          [Function],
          ";     
    padding-bottom: 2;
    padding-left: 10;
    padding-right: 10; 
    margin-bottom: 5px;  
    flex: 1;
    align-items: center;    
",
        ],
      },
      "displayName": "Styled(Component)",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          color="#f00"
          primary={true}
          secondary={true}
          success={true}
        >
          Badge
        </ForwardRef>,
        "color": "#f00",
        "primary": true,
        "secondary": true,
        "success": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Badge",
          "color": "#f00",
          "primary": true,
          "secondary": true,
          "success": true,
        },
        "ref": null,
        "rendered": "Badge",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "    
    flex: 1;
    align-items: center;
    justify-content: center;   
    color: ",
              [Function],
              "; 
    font-size:",
              [Function],
              "; 
    height: 100%;    
",
            ],
          },
          "displayName": "Styled(Component)",
          "propTypes": Object {
            "accessible": [Function],
            "adjustsFontSizeToFit": [Function],
            "allowFontScaling": [Function],
            "disabled": [Function],
            "ellipsizeMode": [Function],
            "minimumFontScale": [Function],
            "nativeID": [Function],
            "numberOfLines": [Function],
            "onLayout": [Function],
            "onLongPress": [Function],
            "onPress": [Function],
            "pressRetentionOffset": [Function],
            "selectable": [Function],
            "selectionColor": [Function],
            "style": [Function],
            "suppressHighlighting": [Function],
            "testID": [Function],
            "textBreakStrategy": [Function],
          },
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "      
    background-color: ",
            [Function],
            ";    
    border: 1px solid ",
            [Function],
            ";        
    width: ",
            [Function],
            ";  
    height: ",
            [Function],
            ";  
    border-radius: ",
            [Function],
            ";     
    padding-bottom: 2;
    padding-left: 10;
    padding-right: 10; 
    margin-bottom: 5px;  
    flex: 1;
    align-items: center;    
",
          ],
        },
        "displayName": "Styled(Component)",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
